R/namespace.R

```diff
@@ -36,14 +41,18 @@ empty_namespace_data <- function() {
 # filter namespace_imports() for S3 generics
 # this loads all imported namespaces
 imported_s3_generics <- function(ns_imports) {
-  is_generic <- vapply(
-    seq_len(nrow(ns_imports)),
-    function(i) {
-      fun_obj <- get(ns_imports$fun[i], envir = asNamespace(ns_imports$pkg[i]))
-      is.function(fun_obj) && is_s3_generic(fun_obj)
-    },
-    logical(1L)
-  )
+  is_generic <- logical(0L)
+
+  if (!is.null(ns_imports)) {
+    is_generic <- vapply(
```

Comment:
how about just using seq_len(NROW(ns_imports))?

Reply:
alternatively, return a 0-row data.frame instead of NULL

Reply:
That's cool! Thanks. 

I also added a comment; lemme know if it looks okay to you.

Reply:
I think too verbose tbh

# NROW for the NULL case of 0-export dependencies, #1503

oughta do